### PR TITLE
Update maintainer docs

### PR DESCRIPTION
**Pull request by tupui**
My mistake, I forgot to update the instructions.

I also just saw the Authors file. Not sure if we should keep it, I added myself there for now then 😅 

Otherwise we might want to fill the changelog in a follow up.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -133,14 +133,12 @@ Following is the process that the development team follows in order to make
...
 1. Document overview of changes since last release in `CHANGELOG.md`
-2. Update the version in the main `__init__.py`.
-3. Build locally using `hatch build`, and verify the content of the artifacts
-4. Submit PR, wait for tests to pass, and merge release into `main`
-5. Tag release with version number and push to SALib repo
-6. Check that release has been deployed to PyPI
-7. Check documentation is built and deployed to readthedocs (http://salib.readthedocs.org)
-8. Check that auto-generated PR is auto-merged on the conda-forge feedstock repo (conda-forge/salib-feedstock)
-9. Update development roadmap on GitHub
+2Build locally using `hatch build`, and verify the content of the artifacts
```
Comment (ConnectedSystems) [inline CONTRIBUTING.md:136]: Typo?

Comment (tupui) [reply CONTRIBUTING.md]: Ark thanks

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -133,14 +133,12 @@ Following is the process that the development team follows in order to make
...
-2. Update the version in the main `__init__.py`.
-3. Build locally using `hatch build`, and verify the content of the artifacts
-4. Submit PR, wait for tests to pass, and merge release into `main`
-5. Tag release with version number and push to SALib repo
-6. Check that release has been deployed to PyPI
-7. Check documentation is built and deployed to readthedocs (http://salib.readthedocs.org)
-8. Check that auto-generated PR is auto-merged on the conda-forge feedstock repo (conda-forge/salib-feedstock)
-9. Update development roadmap on GitHub
+2. Build locally using `hatch build`, and verify the content of the artifacts
+2. Tag release with version number and push to SALib repo
```
Comment (ConnectedSystems) [inline CONTRIBUTING.md:137]: Duplicate list number

**Comment by ConnectedSystems** [comment]
No further action needed for now but I think docs in general need a clean up as there are many duplicate locations where docs are held.

I'll make a release guide page in the doc website and move the instructions there when I have time.

**Pull request merged**